library/core/src/panic/location.rs

```diff
@@ -44,6 +46,42 @@ pub struct Location<'a> {
     _filename: PhantomData<&'a str>,
 }
 
+#[stable(feature = "panic_hooks", since = "1.10.0")]
+impl PartialEq for Location<'_> {
+    fn eq(&self, other: &Self) -> bool {
+        self.file() == other.file() && self.line == other.line && self.col == other.col
```

Comment:
Is it worth flipping the order here? Comparing the filename is expensive, but in 99% of cases the line + column will mismatch on different locations I'd guess.